cocos2d/core/label/CCSGLabelWebGLRenderCmd.js

```diff
@@ -62,32 +66,56 @@ proto._updateDisplayOpacity = function (parentOpacity) {
     this._rebuildLabelSkin();
 };
 
-proto.rendering = function (ctx) {
-    var node = this._node;
-
-    if(node.getString() && (node._labelType === _ccsg.Label.Type.TTF ||
-       node._labelType === _ccsg.Label.Type.SystemFont)){
-        var gl = ctx || cc._renderContext ;
-
-        this._shaderProgram.use();
-        this._shaderProgram._setUniformForMVPMatrixWithMat4(this._stackMatrix);
-
-        ccgl.blendFunc(node._blendFunc.src,node._blendFunc.dst);
-        ccgl.bindTexture2DN(0,this._labelTexture);
-        ccgl.enableVertexAttribs(cc.macro.VERTEX_ATTRIB_FLAG_POS_COLOR_TEX);
-
-        gl.bindBuffer(gl.ARRAY_BUFFER, this._quadWebBuffer);
-
-        if (this._quadDirty) {
-            gl.bufferData(gl.ARRAY_BUFFER, this._quad.arrayBuffer, gl.DYNAMIC_DRAW);
-            this._quadDirty = false;
-        }
-
-        gl.vertexAttribPointer(0, 3, gl.FLOAT, false, 24, 0);                   //cc.macro.VERTEX_ATTRIB_POSITION
-        gl.vertexAttribPointer(1, 4, gl.UNSIGNED_BYTE, true, 24, 12);           //cc.macro.VERTEX_ATTRIB_COLOR
-        gl.vertexAttribPointer(2, 2, gl.FLOAT, false, 24, 16);                  //cc.macro.VERTEX_ATTRIB_TEX_COORDS
+proto.transform = function (parentCmd, recursive) {
+    this.originTransform(parentCmd, recursive);
+
+    var lx = 0, rx = this._labelCanvas.width,
+        by = 0, ty = this._labelCanvas.height,
+        wt = this._worldTransform;
+
+    var vertices = this._vertices;
+    vertices[0].x = lx * wt.a + ty * wt.c + wt.tx; // tl
+    vertices[0].y = lx * wt.b + ty * wt.d + wt.ty;
+    vertices[1].x = lx * wt.a + by * wt.c + wt.tx; // bl
+    vertices[1].y = lx * wt.b + by * wt.d + wt.ty;
+    vertices[2].x = rx * wt.a + ty * wt.c + wt.tx; // tr
+    vertices[2].y = rx * wt.b + ty * wt.d + wt.ty;
+    vertices[3].x = rx * wt.a + by * wt.c + wt.tx; // br
+    vertices[3].y = rx * wt.b + by * wt.d + wt.ty;
+};
 
-        gl.drawArrays(gl.TRIANGLE_STRIP, 0, 4);
+proto.uploadData = function (f32buffer, ui32buffer, vertexDataOffset) {
```

Comment:
This fill all vertex data into the global buffer, the webgl renderer will decide wether this can be batched